### ISSUE TITLE
pulseaudio

### ISSUE THREAD

**Issue by josejp2424** [puppylinux-woof-CE/woof-CE#1740]
I'm following up the development of Woof-CE, but something that should be built in as default is PulseAudio, and leave out aPulse.

Also, as scottman recommended, to have a working bluetooth.

And finally, to make Rox use the icons located on /usr/share/icons.

Standardize Puppy Linux.

**Comment by CollaboratorGCM** [comment]
... GTK3+

WOOFCE does take steps to Standardize Puppy Linux. It also seemingly attempts to leave room for developer choice on a Standard Base.

**Comment by josejp2424** [comment]
> ... GTK3+
> 
> WOOFCE does take steps to Standardize Puppy Linux. It also seemingly attempts to leave room for developer choice on a Standard Base.

yes also migrate to gtk+3

**Comment by peabee** [comment]
Everybody is free to include components such as PulseAudio and GTK3 in their builds if they want....just add them to your build recipe.....but they don't have to be "standardized" as not everybody will need or want them. They are large components and this has to be taken into account when deciding whether to include them.

**Comment by Butterfly** [comment]
> Everybody is free to include components such as PulseAudio and GTK3 in their builds if they want....just add them to your build recipe.....but they don't have to be "standardized" as not everybody will need or want them.

I do *a LOT* of surfing around projects like this. Both PulseAudio and GTK* stir up their share of emotions whenever they're mentioned. There's a long thread in my inbox this very second. Over 40 emails to it. That's pretty regularly a sign that *something* is up with the topic. :smile: 

It's not hard to "imagine".. to predict that some Users might avoid Puppy.. might go elsewhere if either or both were included as a default. When packages such as PulseAudio and GTK* are *not* included, Users know they can **always** personalize their setups with them immediately after an initial install.... :+1:

**Comment by CollaboratorGCM** [comment]
There REALLY should be a General Purpose Puppy standard. AND included should be "standard" removal tool for those who would want to change the "Standard set of subsystems".

Developers "should KNOW" the difference between an "application"  and a subsystem. If they do, then they also know what constitutes a "standard".

Users scream, but developers understand! There is a difference between users and developers that, IMHO, many dont get. GIT is a system whose base supports the work of developers who work as teams.

I applaud what WOOFCE and those who collaborate on it. There "really" is a standard of sorts, but, it needs to be made a little clearer. Further, as time-technology advances, the Standard must also advance periodically to accommodate.

There will **always be** those who will WANT to remain in the past, while **_others are the ones who move us to the present/future_**.

Lastly, if there becomes some way for a clear standard to take a concrete step, one distro app which should be present and obvious to allow ease of progress is to make APPARENT and CLEAR the utility present for those who would want to remove the elements of the running of a "standard" distro. I feel, that users who would NOT want elements can toss them. But, in my past, there are hundreds who will not toss any of them while there are only a very small minority who will.

I ONLY express this, as I can see the Opening Posters request benefit. Having been a part of many teams in my lifetime, BUT I ALSO know that it will take manpower to arrive at a consensus and this could prove problematic in achieving.

**In other words, if it is to be called a "Puppy", what components qualifies it as such?** That in and of itself could be deemed a standard.

@Micko advanced PUPPY with his components including SAMBA, @Jamesbond advanced Puppy with his components for KVM Virtual Machines, @Shino advanced Puppy with FirstRUN, ...  So, the "standard" advanced as time progressed. Puppies today are more robust, expectantly, than in 2005! We have long since moved from those days of 486, tiny drives and tiny RAM along with slow connections for email.

WOOFCE & GIT is the Brave New World for developers.

**Comment by CollaboratorGCM** [comment]
I have been "bothered" by my response. It seems to be either missing something or overlooking something.

Here's the problem, I have with what I shared: 
We all know WOOFCE/WOOFQ/WOOF/etc. allows distro generation. It is flexible enough such that a developer can generate one with merely a few selections OR a developer can generate one with many selections. We have examples of that everywhere in Puppyland of distros that were developed using those distro generators.

This flexibility means that, IMHO, makes it impossible to have a standard....unless, of course, the "Distro Generator Utility" offers a means that provides a single step for creating a Standard PUP. Without that ability for the Generator to open and ask "Do your want a Standard PUP? or, Would you like to Customize a Standard PUP?" I am unclear on how to go forward with a "Puppy LInux Standard". We know that PUPs distros can be generated from various base REpos i.e. APT, DEV, Slack, etc and as such the facilities those master libs manifest are uniquely different at time, so those will "force" generation of a distros whose interiors will have differences.

Does this mean that 'whenever the tool is used to generate a PUP distro, that developers creation is a "standard"?' ... no matter what that developer does to yanks stuff out or put stuff in. It is here that all discussions of a standard get crazy because everyone who uses the tools sees themselves as developing "their standard for their need and their vision".

In My Retrospec, it seems that the ONLY thing I can point to that gives me, personally, a "Standard PUP" is upon arrival at desktop, **the Menus are the same across the PUP distros** and that the tools within, are some expected tools that always there for hardware use and general system services.

Thus, PUPs that do not conform to a some "Standard Menu" structure along with some "Standard utilities" are to be known as a standard while others to be marked as merely based upon being built via use of the WOOF..? And, FOR THOSE WHO DO WOOFY GENERATE, once generated, each developer group should indicate that in additions to producing a Standard PUP, they included these additions.

I can point to some of the developers in this community who do something akin to alerting the users of what is contained. Maybe a standard statement with any additions/deletions would have some easy benefit toward understanding. Distrowatch offers a table that attempts to show a distro's subsystem contents. So that table for distrowatch, IMHO, alludes to subsystem contents which could be deemed a standard for the release of each table's versions listed in the table's column headers.

**FYI**

**Comment by josejp2424** [comment]
> Everybody is free to include components such as PulseAudio and GTK3 in their builds if they want....just add them to your build recipe.....but they don't have to be "standardized" as not everybody will need or want them. They are large components and this has to be taken into account when deciding whether to include them.

**Comment by josejp2424** [comment]
> Everybody is free to include components such as PulseAudio and GTK3 in their builds if they want....just add them to your build recipe.....but they don't have to be "standardized" as not everybody will need or want them. They are large components and this has to be taken into account when deciding whether to include them.


Hello, Peabee.
I know you can put it in the recipe for woof-ce.
But I'd have to be defaulted by now.
Most popular programs use pulseaudio
